config/custom-plugin-monitor.json

```diff
@@ -0,0 +1,32 @@
+{
+  "plugin": "custom",
+  "pluginConfig": {
+    "invoke_interval": "30s",
+    "timeout": "5s",
+    "max_output_length": 80,
+    "concurrency": 3
+  },
+  "source": "ntp-custom-plugin-monitor",
+  "conditions": [
+    {
+      "type": "NTPProblem",
+      "reason": "NTPIsUp",
+      "message": "ntp service is up"
+    }
+  ],
+  "rules": [
+    {
+      "type": "temporary",
+      "reason": "NTPIsDown",
+      "path": "./config/plugin/check_ntp.sh",
+      "timeout": "3s"
+    },
+    {
```

Comment:
Do we run the script twice?
We should emit an event by default for permanent node problem. However, I'm fine with doing that in the future.

Reply:
> Do we run the script twice?

Yes. The reason for doing so is that in this way, we can give users more control on how events and conditions are emitted. When users want an event and condition for a reason, users should declare this explicitly.

Reply:
I think we should generate event for condition change. With that, you shouldn't need 2 rules here.

Reply:
Sometimes, for the same condition, status is not changed, but reason is changed. Without event, people will not even notice that.

Reply:
According to the log condition, when the reason changed, the timestamp and reason will be updated in the node status. I am fine with emitting events when condition changes, but since events has made a stress on ETCD, maybe we should emit events as less as we can. :)